### PR TITLE
Use Temporal signals in Confluence full sync workflow

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR solves https://github.com/dust-tt/dust/issues/3315.

Confluence lacks a feature to limit the resources a token can access. Consequently, Dust must provide a user interface for choosing which spaces should or should not be synchronized within Dust. This situation could complicate our full synchronization process, raising questions about how to handle changes in space selection during an ongoing sync.

This PR resolves the problem by leveraging Temporal's signals to propagate updates to Confluence spaces. If there's an active workflow, it will send a signal to update. If not, it will initiate a full sync workflow.

Example:
```
[confluenceFullSyncWorkflow(confluence-fullsync-22)] [DEBUG] - before entering for ... of: [
  "622596"
]
[confluenceFullSyncWorkflow(confluence-fullsync-22)] [DEBUG] - in for ... of: [
  "622596"
] processing: 622596

-> sending signal for 6127622

[confluenceFullSyncWorkflow(confluence-fullsync-22)] [DEBUG] - in for ... of: [
  "622596",
  "6127622"
] processing: 6127622

[confluenceFullSyncWorkflow(confluence-fullsync-22)] [DEBUG] - after exiting for ... of: [
  "622596",
  "6127622"
]
```

### CODE REVIEW

File: connectors/src/connectors/confluence/temporal/workflows.ts [hunk 1]
```diff
@@ -40,13 +43,25 @@ export async function confluenceFullSyncWorkflow({
...
+  });
+
   const {
     workflowId,
     searchAttributes: parentSearchAttributes,
     memo,
   } = workflowInfo();
 
-  for (const spaceId of spaceIdsToSync) {
+  for (const spaceId of uniqueSpaceIds) {
```
Comment (lasryaric) [inline connectors/src/connectors/confluence/temporal/workflows.ts:64]: Please add a comment here explaning that it actually works.

**Review by lasryaric — CHANGES_REQUESTED** [review]
LGTM minus the comment and the dataSourceConfig object.

File: connectors/src/connectors/confluence/temporal/workflows.ts (hunk 1 above)
Comment (flvndvd) [reply connectors/src/connectors/confluence/temporal/workflows.ts:70]: Comment added. I also added an example of a run in the PR's description.

File: connectors/src/connectors/confluence/temporal/workflows.ts [hunk 2]
```diff
@@ -40,30 +44,55 @@ export async function confluenceFullSyncWorkflow({
...
+          },
+        ],
+        memo,
+      });
+
+      // Remove the processed space from the original set after the async operation.
+      uniqueSpaceIds.delete(spaceId);
+    }
+
+    await sleep(10000);
```
Comment (lasryaric) [inline connectors/src/connectors/confluence/temporal/workflows.ts:95]: do you still need the sleep here?

Comment (flvndvd) [reply connectors/src/connectors/confluence/temporal/workflows.ts]: Optional so I removed it.

**Review by lasryaric — APPROVED** [review]

**Pull request merged**